packages/lib/src/vis/rgb/RgbVis.tsx

```diff
@@ -1,19 +1,20 @@
+import { toTypedNdArray } from '@h5web/shared';
+import type { NdArray } from 'ndarray';
 import { useMemo } from 'react';
 import type { ReactNode } from 'react';
-import { DataTexture, FloatType, RGBFormat, UnsignedByteType } from 'three';
+import { DataTexture, RGBFormat } from 'three';
 
 import styles from '../heatmap/HeatmapVis.module.css';
-import type { Layout } from '../heatmap/models';
+import type { Layout, TextureTypedArray } from '../heatmap/models';
+import { getDims, TEXTURE_TYPE_BY_DTYPE } from '../heatmap/utils';
 import PanZoomMesh from '../shared/PanZoomMesh';
 import VisCanvas from '../shared/VisCanvas';
 import VisMesh from '../shared/VisMesh';
 import { ImageType } from './models';
 import { flipLastDimension } from './utils';
 
 interface Props {
-  value: number[];
-  dims: number[];
-  floatFormat?: boolean;
+  dataArray: NdArray<number[] | TextureTypedArray>; // if `number[]`, assumes uint8: [0, 255]
```

Comment:
I think the comment should rather be on line 36 (where the conversion takes place)